### PR TITLE
Cad refactor

### PR DESCRIPTION
**Pull request by s-oli**
## Description

Address these comments:

### Tasks
- [x] Refactoring:
    - [x] Restructure FreeCADAPI and CadQueryAPI into CADAPI directory (as done [here](https://github.com/Fusion-Power-Plant-Framework/bluemira/tree/je-cook/cadquery/bluemira/codes/cadapi)).
    - [x] Remove ``dotenv`` dependency.
- [x] Explore additional export formats:
    - [x] Particularly ``glTF`` would be nice.
    --> added STL and glTF/GLB
- [x] Missing functions:
    - [x] `move_placement`
    - [?] `make_circle_curve`
    - [?] `make_placement_from_vectors`
    - [?] `plane_from_shape`
    -- Three of the four are FreeCAD-internal helpers used only inside \_freecad/api.py itself, with no external callers in bluemira (make_circle_curve, make_placement_from_vectors) or zero callers anywhere (plane_from_shape).
The CadQuery backend inlines the equivalent OCP calls at the use-sites instead. move_placement was a no-op stub — that's a real bug, fixed in this commit and now covered by test_move_placement_* in the shared backend tests.
- [ ] Explore alternative viewers to Polyscope (as it tessellates):
    - [ ] CQ-editor subset?
- [ ] Testing:
    - [ ] Further edge-cases.
    - [ ] Difficult chains of operations (spline --> offset --> sweep --> cut).
    - [ ] Full EUDEMO run.
## Checklist

I confirm that I have completed the following checks:

- [x] Tests run locally and pass `pytest tests --reactor`
- [x] Code quality checks run locally and pass `pre-commit run --from-ref develop --to-ref HEAD`
- [x] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

**Comment by CoronelBuendia** [comment]
Should this not be targetting https://github.com/Fusion-Power-Plant-Framework/bluemira/pull/4298 ?

**Comment by s-oli** [comment]
> Should this not be targetting #4298 ?

yes -> fixed

File: pyproject.toml [hunk 1]
```diff
@@ -87,20 +86,22 @@ pinned = [
...
-radiation = ["cherab>=1.5"]
+openmc = []  # conda-forge: openmc, openmc_data, tokamak-neutron-source
+radiation = []  # conda-forge: cherab
 dagmc = [
-    "cgal>=6.0.1",
+    # conda-forge: cgal (and bluemira[openmc] would be included if it had any
+    # locked deps — see openmc extra above)
     "fast_ctd @ git+https://github.com/Fusion-Power-Plant-Framework/fast_ctd@0.1.1",
-    "bluemira[openmc]"
 ]
```
Comment (Hudsonbaker4) [inline pyproject.toml]: Is it certain that ``cgal``, ``tokamak-neutron-source``, and ``openmc-data`` can't run without conda dependencies installed?

File: uv.lock
Comment (Hudsonbaker4) [inline uv.lock:1]: We don't currently commit this file.

File: pyproject.toml (hunk 1 above)
Comment (s-oli) [reply pyproject.toml]: All fixed, the extras are reverted to the pre-PR form now that the uv-lock workaround isn't needed. openmc and cherab stay marked as conda-only.

File: uv.lock
Comment (s-oli) [reply uv.lock:1]: uv.lock untracked + gitignored

**Review by Hudsonbaker4 — APPROVED** [review]
